test/e2e/gateway/gatewayapi/gateway_api.go

```diff
@@ -195,6 +195,21 @@ spec:
 			address = net.JoinHostPort(GatewayIP(), "8080")
 		})
 
+		It("should send default static payload for no route", func() {
+			Eventually(func(g Gomega) {
+				// when
+				stdout, _, err := cluster.ExecWithRetries("", "", "",
+					"curl", "-v", "-m", "3", "--fail", "http://"+address)
+
+				// then
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(stdout).ToNot(BeNil())
+
+				g.Expect(stdout).To(ContainSubstring("HTTP/1.1 404 Not Found"))
+				g.Expect(stdout).To(ContainSubstring("This is a Kuma MeshGateway. No routes match this MeshGateway!"))
```

Comment:
I don't think it makes sense to test the exact error message in the test since it's just a static string we set directly, we just want it to be _something_.

Can we do what we (almost) do in `CollectFailure` with `--write-out '%{stderr}%{json}'` and then unmarshal `stderr` as `client.FailureResponse` to check the status code and just check that the body isn't empty via `stdout`?

This is how `CollectFailure` should probably work but no need to refactor that function since it might involve other tests.

Reply:
okay, from what I understand from the above is to make use of `CollectFailure` with `--write-out '%{stderr}%{json}'` curl option and do the rest of the stuff within the test case.

Reply:
I don't mean use `CollectFaulure`, it's just an example of how to do curl with `%{json}`. The idea is we should check the status from `%{json}` but check that the body of the response is not empty.

Otherwise at least just check that the body is not empty instead of checking for the exact body.

Reply:
I'm going to merge this as is and then refactor this/`CollectFailure`.